microsoft/devhome
# PI: Move SnapHelper out of Vertical Bar window and fix snapping bugs.

## Detailed description of the pull request / Additional comments
Move snapping code out to SnapHelper.cs and removing dependency on BarWindowViewModel. This helped fix
1. [Docking and retargeting doesn't work #3034](https://github.com/microsoft/devhome/issues/3034)
2. [PI doesn't set foreground correctly when docked #3030
](https://github.com/microsoft/devhome/issues/3030)
3. [Don't show snap icon when no app is attached](https://github.com/microsoft/devhome/issues/3044)
4. [Also, fixes unexpected elongation of vertical bar window when the user snaps and unsnaps the window multiple times.](https://github.com/microsoft/devhome/issues/3038)

## Validation steps performed
1. Launched PI
3. Attached it to a process
5. Snap bar window to targetapp
6. Move the targetapp, and confirm bar window moves with it.
7. Minimize and restore target app and confirm same actions are taken by bar window.
8. Maximize target app and then confirm bar window is unsnapped to it.
9. Restore target app, then overlap with other apps on it and then snap bar window. Both target app and barwindow come to foreground.
10. Retarget barwindow to a different app and confirm the barwindow is snapped to the new targetapp


## PR checklist
- [ ] Closes #3034 #3030 #3044
